sycl/doc/ControllingSYCLLanguageFeatures.md

```diff
@@ -0,0 +1,195 @@
+# Controlling SYCL/DPC++ language features
+
+This documents aims to describe a mechanism which is used to control (i.e.
+enable or disable) different SYCL/DPC++ language and library features, like
+version of SYCL/DPC++ standard to use, set of extensions to use, etc.
+
+## Controlling SYCL/DPC++ standard version
+
+In order to control the language standard to compile for, the following options
+can be used:
+
+`-sycl-std=<value>`, `--sycl-std=<value>`, `--sycl-std <value>`
+
+SYCL language standard to compile for.
+
+Possible values:
+
+- `2017`, '121', `1.2.1`, `sycl-1.2.1`: corresponds to SYCL 1.2.1
+  specification, see [SYCL Registry] for more details and the specification
+  text.
+
+  Note: setting `-sycl-std` option automatically implies a particular C++
+  standard version to be set.
+
+  **TODO**: do we need to specify which exact version is implied by which
+  `-sycl-std` value? AFAIK, SYCL spec only specifies minimum required C++
+  version, which is C++11 and we set C++17 in our implementation
```

Comment:
I actually like a suggestion from @rolandschulz:

`-sycl-std` doesn't imply any particular C++ standard version to be set and default version is selected (whichever default is for clang). If the default doesn't work as expected, user should use a combination of `-sycl-std` + `-std` to specify both standards.
If some combination of `-sycl-std` and `-std` is not supported, then user will see an error: this could be done by some macro checks within SYCL headers, for example

`-std=sycl-1.2.1` in turn would imply some C++ standard to be configured. Since the spec doesn't tie us to a particular version and only specifies some minimum version, this option would imply _some_ C++ standard, which is at least the minimum version according to the spec.

What do you think?

Reply:
I think it makes sense. Just a question: What happens if `-sycl-std=2020` is used on a clang version that doesn't support C++17 ? I think that should fail. Obviously it cannot happen on DPCPP, but it would be good if at least the flag behavior is documented for all cases so other implementations can do the same.